### ISSUE TITLE
WordPress Connection - Your Ideas & Requests

### ISSUE THREAD

**Issue by easaw** [salesagility/SuiteCRM#2999]
I saw on the Kickstarter video that a WP Portal (or connection) might be in the works for V8 (I realise that was 2014). Now, I'm assuming those plans were for **if** the kickstarter was funded. As it wasn't (sorry to see that), I am thinking that probably half or more of the features won't be implemented for a long time to come, even though it's been a couple of years since that video was released.

There are a few WP plugins laying around, some only on github (albeit most a few years old). I've been working with WP for a few years and would like to try and help in that area.

Basically I'm looking at creating something that connects WP Users to contacts, customers (shoppers) to accounts, products to products and so on.

**Please let me know your ideas for what could/ should be connected (or if SA is already working on that and doesn't need it)**

_I'd also like to help with the Documentation as that was mentioned - however I do see a very good User Guide up at https://suitecrm.com/wiki/ so I assume that's complete..._

**Comment by pgorod** [comment]
There is some discussion of portal features here: https://suitecrm.com/community/forum/suitecrm-7-0-discussion/12510-brainstorming-suitecrm-based-suitecrm-portal

... although it's not related to Wordpress.

I think that any free Wordpress component would be a very valuable addition to SuiteCRM. Thanks for offering to help.

I believe the documentation is still very, very incomplete. The current way to do it (wiki only editable by SalesAgility employees is not really something the community can help with). Anyway, I'm sure they would _love_ to be helped with that; but that still needs some better system to be set up and I guess there hasn't been time to work on that yet.

**Comment by easaw** [comment]
@pgorod who should I speak to on here? Or should I email them?

**Comment by pgorod** [comment]
@easaw they (SalesAgility - SA) are watching you right here, I don't think you need to email anyone. If you can really handle this work, I would say just try to start contributing code to the project - that always gets SA's attention : - )

I would say (although I'm just guessing) that it's safe for you to go ahead and code, probably nobody in SA are working on this, they're too busy with too many other things.

You can post screenshots, demos, even make some Pull Request. That will generate more feedback and attention. I know you probably wanted an architectural discussion before starting to code, but that is not so easy to get around here. Which is understandable, there are so many ideas and issues and a very limited staff to handle everything, plus new features, etc.

And thanks for your willingness to help!

**Comment by chris001** [comment]
Using the REST interface would be good.

**Comment by gunnerman1** [comment]
I am currently coding this up for work. It is not very difficult to get Suite playing with WP using the built-in API. I had a very basic plugin running in an afternoon which created leads in Suite whenever a user filled out a contact form on WP. I first defined a php interface in WP to handle authentication, request building, and handling. Then I have another class which constructs requests to feed to the interface. Of course, this class is integration specific (e.g. how you build a request from WooCommerce and WordPress Core may vary). Do you have a specific integration in mind?

Due to the relative age of the API in Suite, poor documentation, and my general lack of knowledge of REST/HTTP does have me concerned if there are any security issues with it. I would want to do more research/verification and possible upgrades before I transmitted sensitive information in production.

**Comment by easaw** [comment]
@gunnerman1 That sounds good so far, I am interested in getting the Woo integration to work, not for a client etc, but just because I think it's something needed for everyone.

I am not sure if it would be the same as passing things over the same way as with WP.
There are a few plugins on the WP repository that might help us get started. I'd like to test what you've got if you're going to upload it for developer use as I'd be interested in adding to it as I think WP/Woo (well, eCommerce in general) is something the community would like to see.

If you upload it to GitHub I'll be happy to see if I can be helpful and add to it/ try to break it and see what needs working on.

_WP Plugin with form data to lead: https://wordpress.org/plugins/wp-widget-sugarcrm-lead-module/ (unknown price) and https://wordpress.org/plugins/wp-leads-builder-any-crm/ 
(however the pro version is USD$100)_

**Comment by chris001** [comment]
The customer portal running on wordpress:
https://github.com/wp-plugins/sugarcrmsuitecrm-customer-portal

Wordpress-sugarcrm login plugins, form to lead/contact plugins for wordpress to sugarcrm:
https://github.com/search?o=desc&q=wordpress+sugarcrm&s=&type=Repositories&utf8=%E2%9C%93

Prestashop - SuiteCRM - Wordpress 
https://github.com/rohanray/graphql-digital-transformation

**Comment by easaw** [comment]
yeah, $400 for the pro of the 2nd plugin though. no one will pay that...

**Comment by easaw** [comment]
This CodeCanyon actually looks promising:

[Cart2CRM](https://codecanyon.net/item/cart2crm-woocommerce-and-sugarcrm-integration/8234719)

**Comment by chris001** [comment]
It seems there are 4 types of WP (or Joomla) website to SuiteCRM plugins for CRM information sharing.
1. Let users login to both WP and Suite with the same username/password. Sync personal information between WP and SuiteCRM.   For members of the CRM organization so they can use only one username/password for both apps, and it always stays in sync automatically.
2. Lead/Contact website forms for WP, for an inbound marketing web visitor, they fill in their info then the plugin creates a new lead contact or opportunity record in SuiteCRM thru the rest api.
3. Connect WP website and ecommerce addon to SuiteCRM for a new or existing customer, sharing customer account product order billing history information, reports, statistics, when they order items on the WP commerce side, the order data is also created in the CRM side for sale records and order fulfillment shipping accounting taxes follow up support email marketing etc.
4. Portal to let current customers contacts partners to login and view/update their personal/company information, purchase history, calendar, calls, products, create a new support ticket case, upload/download documents, check the knowledge base, ideally every module would be available for the crm admin to set permission whether or not to let the customer interact with their related data records.

The codecanyon one is type 3, sharing sales data between wp ecommerce and crm.

The biztech one from germany is type 3, user information accessing portal.  The free version could be expanded to work with any additional modules you want.

Reference:
* Pro portal details: https://store.biztechconsultancy.com/suitecrm-wordpress-customer-portal-pro.htm
* User facing front end of the portal. http://wpdemo.biztechconsultancy.com/suitecrm_demo_pro/portal-dashboard-gui/
* Back end crm portal. http://sugardemo.biztechconsultancy.com/SuiteCustomerPrortalPro/

**Comment by samus-aran** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [WordPress Connection - Your Ideas & Requests](https://trello.com/c/rdcVuefP/37-wordpress-connection-your-ideas-requests)

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
